srv/src/test/java/my/bookshop/AdminServiceTest.java

```diff
@@ -23,11 +28,15 @@
 
 @RunWith(SpringRunner.class)
 @SpringBootTest
+@AutoConfigureMockMvc
 public class AdminServiceTest {
 
 	@Resource(name = AdminService_.CDS_NAME)
 	private DraftService adminService;
 
+	@Autowired
+	private MockMvc mockMvc;
```

Comment:
The idea of the tests was to show the different layers on which you can test:
1. MockMVC -> Used in CatalogServiceTest
2. local Service API -> Used in AdminServiceTest
3. Unit testing event handlers -> Used in CatalogServiceHandlerTest

I think we should not mix these. You could think about renaming CatalogServiceTest to CatalogServiceITest, as it is more of an integration test and also creating an AdminServiceITest, where you use MockMVC. Alternatively you could also write your test without MockMVC but just using the local Service API.

Reply:
I renamed the CatalogServiceTest to CatalogServiceITest and changed my tests to fit in with the local API tests.